twisted/web/test/test_http.py

```diff
@@ -2470,6 +2471,30 @@ def test_finishProducesLog(self):
                       factory.logFile.getvalue())
 
 
+    def test_requestBodyTimeoutFromFactory(self):
+        """
+        L{HTTPChannel} timeouts whenever data from a request body is not
+        delivered to it in time, even when it gets built from a L{HTTPFactory}.
+        """
+        clock = Clock()
+        factory = http.HTTPFactory(timeout=100, reactor=clock)
+        factory.startFactory()
+        protocol = factory.buildProtocol(None)
+        transport = StringTransport()
+
+        # This is a terrible violation of the abstraction later of
+        # _genericHTTPChannelProtocol, but we need to do it because
+        # policies.TimeoutMixin doesn't accept a reactor on the object.
+        # Maybe we should fix this?
```

Comment:
I don't think that leaving such questions as comments is a good idea.

The question is very good, but I think that it should be part of the PR description or the review request.

Yes. We might find a nicer way to inject the reactor... or use a standard self._reactor or self._scheduler